server/handlers/file_handler.py

```diff
@@ -167,7 +166,7 @@ def get_course(tmp_path):
             assessments.extend(extract_assessments(table))
 
         course_key = get_course_key(pdf)
-        course_info = get_course_info(course_key)
+        course_info = get_course_info(course_key) or {"code": "unknown", "number": "unknown"}
```

Comment:
I won't recommend this failure handling. If user uploads multiple files and multiple files failed to grab course info because course_key is not recognized or doesn't exist, only one entry in the response will represent all the failed ones because the dictionary key `unknown unknown` is going to be overridden below, which causes confusion in front end display.
That's the reason a failure handling was not in the previous PR as failure handling should be taken care by a separate logic.

Reply:
Okay, I'm unfamiliar with the logic/code behind this feature. Would you mind handling it in a future PR or this one?

Reply:
Okay, I'll branch off of this PR and fix this issue